### PR TITLE
plugin-action-button component for consistent styling

### PR DESCRIPTION
**Pull request by kecnry**
### Description

This pull request creates a new `plugin-action-button` UI component for more developer-friendly consistent-styling of buttons in plugins.
* better consistency for coloring by passing `:results_isolated_to_plugin="true/false"`
* update UI style guide
* adds spinner for "Calculate" button in aperture photometry, "Search" in catalogs, and "Apply RGB Presets" in plot options (previously wasn't available because the spinner logic from #2560 was isolated to add-results components)
* changes color of buttons in subset plugin since clicking them affects the app itself

### Change log entry

- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Comment by rosteen** [comment]
This looks good to me so far but I want to do just a little more testing (I did Imviz but not Specviz) in the morning before hitting Approve.

File: jdaviz/components/plugin_table.vue [hunk 1]
```diff
@@ -54,9 +54,8 @@
...
 
     <v-row v-if="clear_table" justify="end">
-      <v-btn
-        color="primary"
-        text
+      <plugin-action-button
+        :results_isolated_to_plugin="true"
         @click="clear_table"
         >Clear Table
       </v-btn>
```
Comment (rosteen) [inline jdaviz/components/plugin_table.vue:61]:
```suggestion
      </plugin-action-button>
```
I think you missed this one.

**Review by rosteen — APPROVED** [review]
Looks good to me after the one suggestion I left above.

File: jdaviz/components/plugin_table.vue (hunk 1 above)
Comment (kecnry) [reply jdaviz/components/plugin_table.vue:61]: woops!!

**Pull request merged**